### ISSUE TITLE
Add access test result in on_exit hook

### ISSUE THREAD

**Issue by lvonk** [elixir-lang/elixir#6938]
As requested in the original post on the [mailing list](https://groups.google.com/d/msg/elixir-lang-core/t4hu5ISQAwA/WDye2UhxAgAJ):

I was looking for a way to access the test result of the failing test in the on_exit hook and found https://github.com/elixir-lang/elixir/issues/2969, but could not find any follow 
up.

What I am trying to achieve is to create a screenshot for each failing browser test (e.g. using Hound), this helps in trying to understand why a certain test failed. 

Basically this could look like:

```elixir
setup context do

  on_exit(context, fn ->
    if context.test_failed
      take_screenshot("./tmp/#{context.test}.png")
    end
  end

  :ok
end
```

Thanks in advance.

**Comment by blatyo** [comment]
The anonymous function would need to accept a context. Otherwise the context would be the one in the setup and could not contain the reason the test failed.

``` elixir
setup context do

  on_exit(fn context ->
    if context.test_failed
      take_screenshot("./tmp/#{context.test}.png")
    end
  end

  :ok
end
```

**Comment by kelvinst** [comment]
Cool feature! Will work on it if no one else is interested! :)

**Comment by josevalim** [comment]
Please go ahead!

**Comment by josevalim** [comment]
I have just realized something. I am not sure if the proposed feature would work because Hound associates the hound session with the test process and by the time on_exit runs the test process is long gone. You may need to have this feature added directly to hound, something like:

    Hound.snapshot_on_error!

And Hound would take care of taking the snapshot when the test errors.

**Comment by lvonk** [comment]
@josevalim Thanks for checking this, I will take this issue to Hound then. @kelvinst thanks for your work 👍 . I really appreciate this was picked up so quickly!
